### PR TITLE
Move site specific details to separate repository

### PR DESCRIPTION
**Pull request by jovial**
This is so we can share a common base and keep the site specific stuff separate. To do this we define all the site specific details in a `recipe`. We then use a config generator to produce the tempest overrides. This helps with code sharing across deployments. See: https://github.com/stackhpc/tempest-recipes for example recipes.

### CODE REVIEW

File: blacklists/helpers/site [hunk 1]
```diff
@@ -0,0 +1,4 @@
+tempest.api.compute.servers.test_create_server.ServersTestManualDisk.test_verify_created_server_vcpus: flavor doesn't match vcpus
```
Comment (markgoddard) [inline blacklists/helpers/site:1]: Looks site-specific

File: blacklists/vm-fixed-network [hunk 2]
```diff
@@ -0,0 +1,12 @@
+tempest.api.compute.admin.test_aggregates.AggregatesAdminTestJSON.test_aggregate_add_host_create_server_with_az: >-
```
Comment (markgoddard) [inline blacklists/vm-fixed-network:1]: Looks site-specific

File: blacklists/vm-floating-ip [hunk 3]
```diff
@@ -0,0 +1,10 @@
+tempest.api.compute.admin.test_aggregates.AggregatesAdminTestJSON.test_aggregate_add_host_create_server_with_az: >-
```
Comment (markgoddard) [inline blacklists/vm-floating-ip:1]: Looks site-specific

File: config_gen/roles/stackhpc.tempest-template/.gitignore [hunk 4]
```diff
@@ -0,0 +1,7 @@
+__pycache__
```
Comment (markgoddard) [inline config_gen/roles/stackhpc.tempest-template/.gitignore:1]: Looks like a separate git repo.

File: tools/tempest-blacklist.py [hunk 5]
```diff
@@ -17,9 +17,14 @@
 
 for regex, reason in regexes.items():
     for line in all_tests:
-        m = re.search(regex, line)
-        if m:
+        m = None
+        try:
+          r = re.compile(regex)
```
Comment (markgoddard) [inline tools/tempest-blacklist.py:22]: nit: indentation looks off

**Comment by jovial** [comment]
Closing in favour of moving the config generator to a separate repository.

**Pull request closed without merging**